magento/magento2
# Filter only visible and enabled products

### Description (*)
In the VisibilityStatusProcessor there was a filter for status and visibility using `joinAttribute` this resulted in getting all products where `status` was greater than 0 and `visibility` was greater than 0. However, the "disabled" status in Magento is `2` and the "not visibile" visibility in Magento is `1`, resulting in all products, including the non-visibile and disabled products, where returned in the GraphQL results.


### Related Pull Requests


### Fixed Issues (if relevant)


### Manual testing scenarios (*)

1. With GraphQL filter products (i.e. related products with an SKU) where there are disabled and enabled products
2. Make sure you return the `status` and `visibility` of the product in the results
3. Check the list of products for disabled or hidden products
4. After this change there should not be any disabled or hidden products, while they were returned before

### Questions or comments


### Contribution checklist (*)


### Resolved issues:
1. [x] resolves magento/magento2#33934: Filter only visible and enabled products
